en/docs/extended_topics/user_op.md

```diff
@@ -1,414 +1,816 @@
-            .Input("y", op.output("out", 0)) /* relu grad op 的一个输入是y，对应的blob为前向op的out(0) */
-            .Input("dy", op.GetGradTensorWithOpOutput("out", 0))  /* 另一个输入dy对应的blob为out_grad */
-            .Output("dx")
-            .Build();
-    /* step 3. 绑定子图的输出blob（relu_grad.out("dx")）与 前向op的input grad blob */
-    op.BindGradTensorWithOpInput(relu_grad_op.output("dx", 0), "in", 0);
-    AddOp(relu_grad_op);  /* step 4. 将子图中新创建的op添加到计算图中 */
-  }
-});
+ctx->DefineOp(op1_name, 
+  [&ctx](user_op::BackwardOpBuilder& builder) {
+    return builder.OpTypeName("multiply")
+        .InputBind("x", ctx->FwOp().input("in", 0)) //multiply.x <- myop.in
+        .InputBind("y", ctx->FwOp().output_grad("out", 0)) //multiply.y <- myop.out的梯度
+        .Output("out")
+        .Build();
+  });
 ```
-- 可能用到的接口介绍：
-  1. `UserOpWrapper`: `.NeedGenGradTensor4OpInput(input_arg_name, index)` 返回一个bool值，判断前向op的输入是否需要生成后向的梯度 `.input(arg_name,index)` 得到输入的logical blob name `.output(arg_name,index)` 得到输出的logical blob name `.attr(attr_name)` 得到op的属性值 `.TensorDesc4ArgNameAndIndex(arg_name, index)` 返回前向op的输入/输出对应的TensorDesc，包含shape、dtype信息 `.GetGradTensorWithOpOutput(output_arg_name, index)` 返回前向op的输出对应的后向梯度blob的logical blob name `.BindGradTensorWithOpInput(logical_blob_name, input_arg_name, index)` 将一个特定的logical blob name与该前向op的输入梯度blob绑定
-  2. `UserOpConfWrapperBuilder`:  （与python端的user_op_builder功能一致，接口类似） `UserOpConfWrapperBuilder(your_op_name)`  构造函数需要输入新构建的op name `.Op(op_type_name)`  指定这个op的type `.Input(arg_name, logical_blob_name)`  可选项，可以调用多次，每次指定一个input_arg_name，同时传入一个logical_blob_name，表明这个input arg name对应的blob。如果该input_arg_name对应多个输入blob，则调用`.Input()`的顺序就是其对应的index顺序 `.Output(arg_name, num)`  可选项，可调用多次，每次指定一个`output_arg_name`实际对应的输出blob的数量，也可以调用 `.Output(arg_name)`，表示`num = 1` `.Attr(attr_name, val)` 可选项，可调用多次，每次指定一个attr属性的属性名称和参数值，表示对这个attr赋值为val `.Build()`  返回一个UserOpConfWrapper，表示你构建完毕的新op
-  3. `UserOpConfWrapper`: `.input(arg_name,index)` 得到输入的logical blob name `.output(arg_name,index)` 得到输出的logical blob name `.attr(attr_name)` 得到op的属性值
-  4. `AddOp`: 输入参数是一个UserOpConfWrapper
+
+In this function, we call `Build` to build a reverse op for computing `x*dy`.
+The purpose of each operator is as follows:
+
+* `OpTypeName("multiply")` specifies the `op_type_name` of an op that is used to help us compute the reverse gradient.
+
+* `InputBind(arg_name, blob)` binds the input `arg_name` of `multiply` to the specified `blob` and can be called for multiple times. If the `arg_name` corresponds to multiple  blob which means the order of `Input` is the order of the corresponding index.
+
+* `Output(arg_name, num)` Specifies the number of output blobes that actually correspond to the `arg_name` which defaults to 1 if `num` is not filled in.
+
+* `Attr(attr_name, val) ` sets the value of the attribute which same in the registration.
+
+* After completing settings in `Build()`. Then complete the construction of the reverse op by calling `Build`.
```

Comment:
Calling `Build()` after above configuration, then the construction of the reverse op is completed.